sencha-workspace/SlateDemonstrationsStudent/app/controller/Dashboard.js

```diff
@@ -250,6 +256,9 @@ Ext.define('SlateDemonstrationsStudent.controller.Dashboard', {
 
         Ext.resumeLayouts(true);
 
+        // show/hide components based on enrollment
+        [competenciesSummary, recentProgress, levelsLegend, cardsCt ].forEach(cntr => cntr.setVisible(studentEnrolled));
```

Comment:
For all these components, we use a "subcomponent slot" strategy where the config system can be used to tell the main component which subcomponents should be hidden/visible without knowing the details of how hiding/showing them should work. We can get rid of all the added refs, and instead of getting a ref to each subcomponent we just want to do `mainComponent.setSubComponent(shouldBeVisible)`

Reply:
@themightychris I implemented this here:  https://github.com/SlateFoundation/slate-cbl/commit/a5c33a6c3832f6061ee0a9511bf9502838bee16c